### ISSUE TITLE
Dependency with custom variable for parent

### ISSUE THREAD

**Issue by dgoetz** [Icinga/icingaweb2-module-director#1208]
## Expected Behavior
In addition to the current behavior it would be useful to be able to set the parent host and/or service to a custom variable instead of an object.

Something like this:
```
apply Dependency "gateway" to Host {
  import "Host Dependency Template"

  parent_host_name = host.vars.gateway
  assign where host.vars.gateway
}
```

Macro syntax `parent_host_name = "$host.vars.gateway$"` is not allowed here.

## Current Behavior
Only objects are allowed.

## Context
Allowing something like this will allow to reduce the number of dependencies to maintain.

## Your Environment
* Director version (System - About): Git Master 6980544
* Icinga Web 2 version and modules (System - About): 2.4.1
* Icinga 2 version (`icinga2 --version`): 2.7.1
* Operating System and version: RHEL 7
* Webserver, PHP versions:  Apache 2.4, PHP 5.4.45

**Comment by migodev** [comment]
I also missing this Feature.

**Comment by jpkaiser** [comment]
+1
need this too..

**Comment by avik78** [comment]
+1 

this feature would be really appreciated

**Comment by cfahnemann** [comment]
+1
This is exactly our required use case:
https://www.icinga.com/docs/icinga2/latest/doc/03-monitoring-basics/#apply-dependencies-based-on-custom-attributes

**Comment by themelle** [comment]
+1

this would - for the first time - make dependency apply rules work in a useful and intuitive way

**Comment by phil-or** [comment]
+1

**Comment by stevie-sy** [comment]
+1

**Comment by dgoetz** [comment]
Can everyone please stop spamming with +1? If you just want to be notified hit subscribe, if you want +1 use the emoji on the initial comment. Spamming developers, issue creators and others is not helping, just annoying.

**Comment by Thomas-Gelf** [comment]
I cannot promise this for 1.6, but I'm pinning it right now to have a reminder. Will be considered for v1.6.0 in case there is enough time

**Comment by nbuchwitz** [comment]
@Thomas-Gelf did you already started with something? If not I'd be happy to have a look into it, as I could really use it at a customers.

**Comment by Thomas-Gelf** [comment]
This should already work since quite some time, I fixed a rendering-related bug recently (#2088). Form element description reads:

> The parent host. **You might want to refer Host Custom Variables via $host.vars.varname$**
